### ISSUE TITLE
OpenCV 3 macOS 10.12 Sierra QTKit not found build issue

### ISSUE THREAD

**Issue by kushalj1997** [opencv/opencv#7164]
#System information

- OpenCV => 3.1
- Operating System / Platform => macOS Sierra 10.12 (16A304a)
- Compiler => Xcode

https://github.com/Homebrew/homebrew-science/issues/4104

**Comment by mshabunin** [comment]
Duplicate of #6913

**Comment by kushalj1997** [comment]
Not a duplicate, he has COMPILE issues. I'm having general build issues to get a new install of OpenCV on Sierra. OpenCV is not installed yet on my Mac; he has it already installed then installed macOS over it. The "solution" he uses is a coding one, and I cannot use that solution since I don't even have OpenCV built for my computer.

**Comment by mshabunin** [comment]
@kushalj1997 , the problem is the same - QtKit has been removed from recent OSX, and OpenCV does not support this OS yet.

You can try to use `with-quicktime` brew option as workaround.
From opencv3.rb:
```
...
option "with-quicktime", "Use QuickTime for Video I/O instead of QTKit"
...
```

**Comment by kushalj1997** [comment]
`brew install opencv3 --with-contrib --with-quicktime --with-ffmpeg`
```
==> Auto-updated Homebrew!
Updated Homebrew from 72846fa to bf87267.
Updated 3 taps (homebrew/core, homebrew/python, homebrew/science).
==> New Formulae
homebrew/python/aeneas                   libdivecomputer                        
==> Updated Formulae
homebrew/python/mpi4py     homebrew/python/scapy      homebrew/science/zoltan  
homebrew/python/pillow     homebrew/science/petsc     joe                      
Warning: You are using OS X 10.12.
We do not provide support for this pre-release version.
You may encounter build failures or other breakages.
Please create pull-requests instead of filing issues.
==> Installing opencv3 from homebrew/science
==> Installing dependencies for homebrew/science/opencv3: openexr
==> Installing homebrew/science/opencv3 dependency: openexr
==> Using the sandbox
==> Downloading https://savannah.nongnu.org/download/openexr/openexr-2.2.0.tar.g
==> Downloading from http://nongnu.askapache.com//openexr/openexr-2.2.0.tar.gz
######################################################################## 100.0%
==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/f1a3e
######################################################################## 100.0%
==> Patching
==> Applying 64bit_types.patch
patching file IlmImf/ImfFastHuf.cpp
patching file IlmImf/ImfFastHuf.cpp
==> ./configure --prefix=/usr/local/Cellar/openexr/2.2.0
==> make install
🍺  /usr/local/Cellar/openexr/2.2.0: 131 files, 11.2M, built in 3 minutes 38 seconds
==> Installing homebrew/science/opencv3
==> Downloading https://github.com/Itseez/opencv/archive/3.1.0.tar.gz
==> Downloading from https://codeload.github.com/opencv/opencv/tar.gz/3.1.0
######################################################################## 100.0%
==> Downloading https://github.com/Itseez/opencv/commit/a2bda999211e8be9fbc5d400
==> Downloading from https://github.com/opencv/opencv/commit/a2bda999211e8be9fbc
######################################################################## 100.0%
==> Downloading https://github.com/Itseez/opencv/commit/c7bdbef5042dadfe032dfb5d
==> Downloading from https://github.com/opencv/opencv/commit/c7bdbef5042dadfe032
######################################################################## 100.0%
==> Downloading https://github.com/Itseez/opencv/commit/cdb9c60dcb65e04e7c0bd6be
==> Downloading from https://github.com/opencv/opencv/commit/cdb9c60dcb65e04e7c0
######################################################################## 100.0%
==> Patching
==> Applying a2bda999211e8be9fbc5d40038fdfc9399de31fc.diff
patching file modules/videoio/src/cap_qtkit.mm
==> Applying c7bdbef5042dadfe032dfb5d80f9b90bec830371.diff
patching file modules/core/include/opencv2/core/opencl/runtime/autogenerated/opencl_gl.hpp
Hunk #1 succeeded at 45 (offset 4 lines).
Hunk #2 succeeded at 58 (offset 4 lines).
patching file modules/core/include/opencv2/core/opencl/runtime/autogenerated/opencl_gl_wrappers.hpp
patching file modules/core/src/opencl/runtime/autogenerated/opencl_gl_impl.hpp
patching file modules/core/src/opencl/runtime/generator/common.py
patching file modules/core/src/opencl/runtime/generator/template/opencl_gl.hpp.in
Hunk #1 succeeded at 12 (offset 4 lines).
patching file modules/core/src/opencl/runtime/generator/template/opencl_gl_impl.hpp.in
patching file modules/core/src/opencl/runtime/generator/template/opencl_gl_wrappers.hpp.in
patching file modules/core/src/opencl/runtime/opencl_core.cpp
Hunk #1 succeeded at 283 (offset -3 lines).
Hunk #2 succeeded at 305 (offset -3 lines).
patching file modules/core/src/opengl.cpp
==> Applying cdb9c60dcb65e04e7c0bd6bef9b86841191c785a.diff
patching file samples/gpu/CMakeLists.txt
==> Downloading https://github.com/Itseez/opencv_contrib/archive/3.1.0.tar.gz
==> Downloading from https://codeload.github.com/opencv/opencv_contrib/tar.gz/3.
######################################################################## 100.0%
==> Downloading https://raw.githubusercontent.com/Itseez/opencv_3rdparty/81a6760
######################################################################## 100.0%
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG
==> make
Last 15 lines from /Users/jollygama/Library/Logs/Homebrew/opencv3/02.make:
[ 47%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_mjpeg_decoder.cpp.o
cd /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio && /usr/local/Library/Homebrew/shims/super/clang++   -DCVAPI_EXPORTS -DVIDEOIO_EXPORTS -D__OPENCV_BUILD=1 -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/3rdparty/ippicv/unpack/ippicv_osx/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild -isystem /usr/local/include/eigen3 -isystem /usr/local/Cellar/ffmpeg/3.1.2/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/core/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgproc/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgcodecs/include  -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -fno-omit-frame-pointer -msse -msse2 -mavx -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated-declarations -DNDEBUG  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -fPIC   -o CMakeFiles/opencv_videoio.dir/src/cap_images.cpp.o -c /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src/cap_images.cpp
cd /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio && /usr/local/Library/Homebrew/shims/super/clang++   -DCVAPI_EXPORTS -DVIDEOIO_EXPORTS -D__OPENCV_BUILD=1 -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/3rdparty/ippicv/unpack/ippicv_osx/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild -isystem /usr/local/include/eigen3 -isystem /usr/local/Cellar/ffmpeg/3.1.2/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/core/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgproc/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgcodecs/include  -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -fno-omit-frame-pointer -msse -msse2 -mavx -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated-declarations -DNDEBUG  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -fPIC   -o CMakeFiles/opencv_videoio.dir/src/cap_mjpeg_decoder.cpp.o -c /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src/cap_mjpeg_decoder.cpp
cd /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio && /usr/local/Library/Homebrew/shims/super/clang++   -DCVAPI_EXPORTS -DVIDEOIO_EXPORTS -D__OPENCV_BUILD=1 -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/3rdparty/ippicv/unpack/ippicv_osx/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild -isystem /usr/local/include/eigen3 -isystem /usr/local/Cellar/ffmpeg/3.1.2/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/core/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgproc/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgcodecs/include  -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -fno-omit-frame-pointer -msse -msse2 -mavx -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated-declarations -DNDEBUG  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -fPIC   -o CMakeFiles/opencv_videoio.dir/src/cap.cpp.o -c /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src/cap.cpp
[ 47%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o
cd /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio && /usr/local/Library/Homebrew/shims/super/clang++   -DCVAPI_EXPORTS -DVIDEOIO_EXPORTS -D__OPENCV_BUILD=1 -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/3rdparty/ippicv/unpack/ippicv_osx/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild -isystem /usr/local/include/eigen3 -isystem /usr/local/Cellar/ffmpeg/3.1.2/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/core/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgproc/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgcodecs/include  -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -fno-omit-frame-pointer -msse -msse2 -mavx -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated-declarations -DNDEBUG  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -fPIC   -o CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o -c /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src/cap_ffmpeg.cpp
[ 47%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_qt.cpp.o
cd /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio && /usr/local/Library/Homebrew/shims/super/clang++   -DCVAPI_EXPORTS -DVIDEOIO_EXPORTS -D__OPENCV_BUILD=1 -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/3rdparty/ippicv/unpack/ippicv_osx/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild -isystem /usr/local/include/eigen3 -isystem /usr/local/Cellar/ffmpeg/3.1.2/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/macbuild/modules/videoio -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/core/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgproc/include -I/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/imgcodecs/include  -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -fno-omit-frame-pointer -msse -msse2 -mavx -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated-declarations -DNDEBUG  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -fPIC   -o CMakeFiles/opencv_videoio.dir/src/cap_qt.cpp.o -c /tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src/cap_qt.cpp
/tmp/opencv3-20160825-8502-10rqva3/opencv-3.1.0/modules/videoio/src/cap_qt.cpp:63:10: fatal error: 'QuickTime/QuickTime.h' file not found
#include <QuickTime/QuickTime.h>
         ^
1 error generated.
make[2]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_qt.cpp.o] Error 1
make[1]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/all] Error 2
make: *** [all] Error 2

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/brew):
  https://github.com/Homebrew/homebrew-science/issues

These open issues may also help:
opencv3: fix build with vtk https://github.com/Homebrew/homebrew-science/pull/3749
opencv3 fails to create '/usr/local/opt/opencv3/python3.5' folder with 'without numpy' and 'with python3' options https://github.com/Homebrew/homebrew-science/issues/4082
opencv3 with python3 doesn't have python3.5 in the lib folder in the Cellar folder https://github.com/Homebrew/homebrew-science/issues/3939
opencv3 python3 wrappers can not be generated https://github.com/Homebrew/homebrew-science/issues/3302
OpenCV and OpenCV3 fail to build with ximea camera support https://github.com/Homebrew/homebrew-science/issues/3395
opencv3 missing xz dependency in linuxbrew https://github.com/Homebrew/homebrew-science/issues/2642
opencv3 PYTHON_PATH and other variables set incorrectly in some cases https://github.com/Homebrew/homebrew-science/issues/2846
OpenCV3 failed to build with opengl on OSX https://github.com/Homebrew/homebrew-science/issues/3383
opencv3 misconfigures cmake because brew incorrectly sets $PYTHONPATH https://github.com/Homebrew/homebrew-science/issues/3329
opencv3 --with-contrib --with-ffmpeg --HEAD build error with Mac OS X El Capitan 10.11 https://github.com/Homebrew/homebrew-science/issues/3367
opencv3 failed to build on OS X 10.10.5 https://github.com/Homebrew/homebrew-science/issues/2733
Fail to install opencv3 --with-python3 --with-contrib on El Capitan https://github.com/Homebrew/homebrew-science/issues/3051
opencv3 build fails without clear error message on OSX 10.10.5 https://github.com/Homebrew/homebrew-science/issues/3119
Warning: You are using OS X 10.12.
We do not provide support for this pre-release version.
You may encounter build failures or other breakages.
Please create pull-requests instead of filing issues.
```

I ran the option with-quicktime, and this is the result.

**Comment by mshabunin** [comment]
It means that QuickTime has been removed too. So you need to wait until fix is merged: #7159
